File: app/src/main/java/com/erikzuo/portfolioandroidapp/utils/rx/SchedulerProvider.java
package com.erikzuo.portfolioandroidapp.utils.rx;

import io.reactivex.Scheduler;

/**
 * Created by YifanZuo on 3/2/18.
 */

public interface SchedulerProvider {

    Scheduler ui();

    Scheduler computation();

    Scheduler io();

}

File: app/src/main/java/com/erikzuo/portfolioandroidapp/utils/rx/SchedulerProvider.kt
package com.erikzuo.portfolioandroidapp.utils.rx

import io.reactivex.Scheduler

/**
 * Created by YifanZuo on 3/2/18.
 */

interface SchedulerProvider {

    fun ui(): Scheduler

    fun computation(): Scheduler

    fun io(): Scheduler

}
